.github/actions/gen-docs/action.yaml

```diff
@@ -0,0 +1,56 @@
+name: "Generate all files related to documentation"
+
+description: "Generates all files related to documentation."
+
+inputs:
+  username:
+    description: "GitHub username"
+    required: true
+  email:
+    description: "GitHub email"
+    required: true
+  token:
+    description: "GitHub Token (must be a PAT for repository dispatch)"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install python and set up Poetry
+      uses: bakdata/ci-templates/actions/python-setup-poetry@v1.5.3
+      with:
+        poetry-version: "1.3.2"
```

Comment:
Maybe it is a good idea to add this as an input parameter.  Incase we change that poetry version in the CI workflow we don't forget it here